### DISCUSSION TITLE
Deactivating ActivityPub Plugin >> What happens with your WordPress federation?

### DISCUSSION THREAD

**Discussion by p37307** [Automattic/wordpress-activitypub#784] [Q&A]
What happens in the Fediverse if you deactivate the WordPress ActivityPub plugin?

When you deactivate the ActivityPub plugin, does it send a message to other servers in the Fediverse that you are no longer part of it? Does it allow a clean exit? etc.

Mastodon has a self-destruct option in their CLI that erases the _"server from the federation by broadcasting account Delete activities to all known other servers. This allows a “clean exit” from running a Mastodon server, as it leaves next to no cache behind on other servers."_ 

What happens if you reactivate it at a later time?

I've looked and can't really find anything about it, although, I wouldn't be surprised if I missed it in my searching.

**Comment by rghedin** [comment]
This is an important question. @pfefferle, can you help us?

**Comment by pfefferle** [comment]
I agree that this is important!

The short answer is: We are working on it: https://github.com/Automattic/wordpress-activitypub/pull/552

I will try to go into more details, why this is not trivial on WordPress, when I am back at the computer!

**Reply by p37307** [reply]
Thanks for the update,  @pfefferle

**Comment by pfefferle** [comment]
Some more thoughts on that:

We thought about that missing feature a lot in the last few weeks and it is not trivial.

The main issue: explicit vs. implicit

It would be easy, to simply send a delete, if you remove the post-credentials from a user or if you deactivate the plugin (the last one might still be tricky, because if the plugin is deactivated, we can't send something any more, but there might be a solution with postponing the deactivation a bit), but, as far as I know, deletes are irrevocable.

So if we send a self-destruct, after you deactivated the plugin, it might be possible, that this happens on accident, because you wanted to debug something or hit the deactivate by accident. The same with the users, if you change credentials to test something, you do not expect the user to be deleted.

That said: I think we need a way for users to delete things (for example a user or the whole blog), by explicitly clicking "delete" buttons.

I hope this makes sense.

As I already mentioned in the comment above, we are working on a first step, but are fighting with some issues, like the signatures https://github.com/Automattic/wordpress-activitypub/pull/552/files#r1602741399

**Reply by p37307** [reply]
Any progress on this or [552](https://github.com/Automattic/wordpress-activitypub/pull/552)? I have a domain that host(s) Wordpress that has been in limbo for a while because I want to send a request to the ActivityPub Fediverse to delete the federated Wordpress Blog. We had some discussions on Mastodon here: https://oldfriends.live/@paul/112683485805574954

**Comment by danimrich** [comment]
A manual way of sending a "delete" (without deactivating the plugin) would be very helpful. (My apologies if this is slightly offtopic.)

Here's my problem: When trying out the plugin, I played around a bit before I decided to use @(domain)@(domain) as blog profile and not use any author profiles. When I now search for (domain) on mastodon.social, I get a few results:

- @ blog@(domain) - former blog profile.
- @ (author)@(domain) - former author profile.
- @ application@(domain) - a profile marked as bot. I don't know what it does. 
- @ (domain)@(domain) - blog profile. Active.

Ideally I'd like to find ONLY the active blog profile. It would be nice if it were possible to manually remove the three other profiles since they look like active profiles, but actually aren't. I haven't used them for a while, yet they still seem to be cached on mastodon.social.

**Reply by pfefferle** [reply]
You can use the CLI for that. Use `wp activitypub --help` to see the required syntax

**Reply by danimrich** [reply]
Oh, thanks for the quick reply! I didn't know that this was possible.

**Reply by rghedin** [reply]
>You can use the CLI for that. Use wp activitypub --help to see the required syntax

Can't find this option, @pfefferle (AP v7.1.0):

<img width="540" height="168" alt="image" src="https://github.com/user-attachments/assets/4acad150-f634-4313-9d2a-a3eb510e0165" />

**Comment by rghedin** [comment]
There's [some action going on here](https://github.com/Automattic/wordpress-activitypub/milestone/3), including a WIP PR (#2035) for `Delete` activity. Maybe landing on v7.3…? 👀
